aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1158,12 +1169,15 @@ Tensor& argmax_out(const Tensor& self, c10::optional<int64_t> dim, bool keepdim,
     }
     in = self;
   } else {
+    TORCH_CHECK_INDEX(self.numel() != 0, "argmax_out(): Expected reduction dim for self.numel() == 0.");
```

Comment:
Suggested tweak for this error message:

"Expected dim to be specified for an empty input tensor."

Reply:
Doesn't `self.numel()` point specifically to the problem that `numel` is 0? This way the user can simply query for `numel()` on their object and quickly figure out where the problem comes from.

If we specify `empty tensor` they might check for `tensor.size()` which is a tuple and requires searching for the 0 dimension in the tensor.